packages/protocol/contracts/governance/LockedGold.sol

```diff
@@ -5,777 +5,438 @@ import "openzeppelin-solidity/contracts/math/SafeMath.sol";
-  struct Commitment {
-    uint128 value;
-    uint128 index;
+  struct MustMaintain {
+    uint256 value;
+    uint256 timestamp;
   }
 
-  struct Commitments {
-    // Maps a notice period in seconds to a Locked Gold commitment.
-    mapping(uint256 => Commitment) locked;
-    // Maps an availability time in seconds since epoch to a notified commitment.
-    mapping(uint256 => Commitment) notified;
-    uint256[] noticePeriods;
-    uint256[] availabilityTimes;
+  struct Authorizations {
+    address voting;
+    address validating;
   }
 
-  struct Account {
-    bool exists;
-    // The weight of the account in validator elections, governance, and block rewards.
-    uint256 weight;
-    // Each account may delegate their right to receive rewards, vote, and register a Validator or
-    // Validator group to exactly one address each, respectively. This address must not hold an
-    // account and must not be delegated to by any other account or by the same account for any
-    // other purpose.
-    address[3] delegates;
-    // Frozen accounts may not vote, but may redact votes.
-    bool votingFrozen;
-    // The timestamp of the last time that rewards were redeemed.
-    uint96 rewardsLastRedeemed;
-    Commitments commitments;
-  }
-
-  // TODO(asa): Add minNoticePeriod
-  uint256 public maxNoticePeriod;
-  uint256 public totalWeight;
-  mapping(address => Account) private accounts;
-  // Maps voting, rewards, and validating delegates to the account that delegated these rights.
-  mapping(address => address) public delegations;
-  // Maps a block number to the cumulative reward for an account with weight 1 since genesis.
-  mapping(uint256 => FixidityLib.Fraction) public cumulativeRewardWeights;
-
-  event MaxNoticePeriodSet(
-    uint256 maxNoticePeriod
-  );
-
-  event RoleDelegated(
-    DelegateRole role,
-    address indexed account,
-    address delegate
-  );
-
-  event VotingFrozen(
-    address indexed account
-  );
-
-  event VotingUnfrozen(
-    address indexed account
-  );
-
-  event NewCommitment(
-    address indexed account,
-    uint256 value,
-    uint256 noticePeriod
-  );
-
-  event CommitmentNotified(
-    address indexed account,
-    uint256 value,
-    uint256 noticePeriod,
-    uint256 availabilityTime
-  );
-
-  event CommitmentExtended(
-    address indexed account,
-    uint256 value,
-    uint256 noticePeriod,
-    uint256 availabilityTime
-  );
-
-  event Withdrawal(
-    address indexed account,
-    uint256 value
-  );
-
-  event NoticePeriodIncreased(
-    address indexed account,
-    uint256 value,
-    uint256 noticePeriod,
-    uint256 increase
-  );
+  struct PendingWithdrawal {
+    uint256 value;
+    uint256 timestamp;
+  }
 
-  function initialize(address registryAddress, uint256 _maxNoticePeriod) external initializer {
-    _transferOwnership(msg.sender);
-    setRegistry(registryAddress);
-    maxNoticePeriod = _maxNoticePeriod;
+  struct Balances {
```

Comment:
document, does account total balance be the sum of the three fields?
From reading this, the one that not so clear is `requirements`.  what does it express... who sets it. It becomes clear after you read the whole code, but it good to have a commetn here